livekit-api/livekit/api/agent_dispatch_service.py

```diff
@@ -0,0 +1,108 @@
+import aiohttp
+from typing import Optional
+from livekit.protocol import agent_dispatch as proto_agent_dispatch
+from ._service import Service
+from .access_token import VideoGrants
+
+SVC = "AgentDispatchService"
+
+
+class AgentDispatchService(Service):
+    """Manage agent dispatches. Service APIs require roomAdmin permissions.
+
+    An easier way to construct this service is via LiveKitAPI.agent_dispatch.
+    """
+
+    def __init__(
+        self, session: aiohttp.ClientSession, url: str, api_key: str, api_secret: str
+    ):
+        super().__init__(session, url, api_key, api_secret)
+
+    async def create_dispatch(
+        self, req: proto_agent_dispatch.CreateAgentDispatchRequest
+    ) -> proto_agent_dispatch.AgentDispatch:
+        """Create an explicit dispatch for an agent to join a room.
+
+        To use explicit dispatch, your agent must be registered with an `agentName`.
+
+        Args:
+            req (CreateAgentDispatchRequest): Request containing dispatch creation parameters
+
+        Returns:
+            AgentDispatch: The created agent dispatch object
+        """
+        return await self._client.request(
+            SVC,
+            "CreateDispatch",
+            req,
+            self._auth_header(VideoGrants(room_admin=True, room=req.room)),
+            proto_agent_dispatch.AgentDispatch,
+        )
+
+    async def delete_dispatch(
+        self, dispatch_id: str, room_name: str
+    ) -> proto_agent_dispatch.AgentDispatch:
+        """Delete an explicit dispatch for an agent in a room.
+
+        Args:
+            dispatch_id (str): ID of the dispatch to delete
+            room_name (str): Name of the room containing the dispatch
+
+        Returns:
+            AgentDispatch: The deleted agent dispatch object
+        """
+        return await self._client.request(
+            SVC,
+            "DeleteDispatch",
+            proto_agent_dispatch.DeleteAgentDispatchRequest(
+                dispatch_id=dispatch_id,
+                room=room_name,
+            ),
+            self._auth_header(VideoGrants(room_admin=True, room=room_name)),
+            proto_agent_dispatch.AgentDispatch,
+        )
+
+    async def list_dispatch(
+        self, room_name: str
+    ) -> list[proto_agent_dispatch.AgentDispatch]:
+        """List all agent dispatches in a room.
+
+        Args:
+            room_name (str): Name of the room to list dispatches from
+
+        Returns:
+            list[AgentDispatch]: List of agent dispatch objects in the room
+        """
+        res = await self._client.request(
+            SVC,
+            "ListDispatch",
+            proto_agent_dispatch.ListAgentDispatchRequest(room=room_name),
+            self._auth_header(VideoGrants(room_admin=True, room=room_name)),
+            proto_agent_dispatch.ListAgentDispatchResponse,
+        )
+        return list(res.agent_dispatches)
+
+    async def get_dispatch(
+        self, dispatch_id: str, room_name: str
```

Comment:
For consistency, wdyt of using `proto_agent_dispatch.ListAgentDispatchRequest` as the parameter. (not a big deal otherwise)

Reply:
I think here i'm mainly trying to make the API easier to understand. it's always kind of confusing seeing a `List*` API for when you are looking to retrieve a single element.

Reply:
For other services, we don't expose any get_ method.
Only list (e.g [egress](https://github.com/livekit/python-sdks/blob/e18187079b4706d78ed70fcecd5124f66eaa5e9c/livekit-api/livekit/api/egress_service.py#L92))

Maybe that's fine but adding new parameters in the future may be a pain. So if we go into this path, can you make them kwarg

Reply:
we didn't have it for the other services, and I think that's a shortcoming. optimizing for user clarity, `get` > `list` when getting a single entry.

I'm not sure that it makes sense to make the required args as kwargs. what is the motivation there? we can always add kwargs for optional args as kwargs.

Reply:
It makes the code more explicit (e.g which position is the room name, first or second?)

Reply:
I generally use positional argument when there is only one **main** arguement. Tho no strong opinion

Reply:
I see what you mean, especially when it gets longer, but 2 seems very straight forward IMHO.

Reply:
fwiw it's still possible to treat positional arguments as named in python. this just allows you to do either/or

Reply:
I generally max to 1, but no strong opinion, lgtm